### PR TITLE
feat(masthead): makes search and profile optional

### PR DESCRIPTION
**Pull request by kennylam**
### Related Ticket(s)

#101 

### Description

This change allows the Profile and Search components in the masthead to be optional. All masthead components (except the IBM logo) are now optional.

### Changelog

**New**

- Search and profile are now optional

**Changed**

- `Masthead` and `DotcomShell` now share the same knobs for `Masthead` component

### CODE REVIEW

File: packages/react/src/components/Masthead/Masthead.js [hunk 1]
```diff
@@ -138,51 +138,59 @@ const Masthead = ({ navigation, ...mastheadProps }) => {
...
-                {navigation !== false ? (
+                {navigation && (
                   <MastheadTopNav
                     {...mastheadProps}
                     navigation={mastheadData}
                   />
-                ) : null}
-                <MastheadSearch />
+                )}
+                {mastheadProps.hasSearch && <MastheadSearch />}
```
Comment (jeffchew) [inline packages/react/src/components/Masthead/Masthead.js]: @kennylam I couldn't find defaultProps for these, these should be true by default right?

Comment (kennylam) [reply packages/react/src/components/Masthead/Masthead.js]: @jeffchew Doh! Yes you are right 👍

File: packages/react/src/components/Masthead/Masthead.js [hunk 2]
```diff
@@ -200,4 +208,13 @@ Masthead.propTypes = {
   mastheadProp: PropTypes.object,
 };
 
+/**
+ * @property defaultProps
+ * @type {{hasProfile: boolean, hasSearch: boolean}}
+ */
+Masthead.defaultProps = {
+  hasProfile: true,
+  hasSearch: true,
```
Comment (jeffchew) [inline packages/react/src/components/Masthead/Masthead.js]: @kennylam propTypes need to be added too...

**Comment by kennylam** [comment]
@oliviaflory This PR addresses #158 also.

**Pull request merged**